R/zchunk_LB1091.ag_GrossTrade.R

```diff
@@ -1,5 +1,48 @@
 # Copyright 2019 Battelle Memorial Institute; see the LICENSE file.
 
+#' module_data_FAO_BilateralTrade
+#'
+#' Read the FAO_BilateralTrade trade from file and reduce / summarize as soon as possible to
+#' keep it's memory footprint small
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{FAO_BilateralTrade}.
+#' @importFrom dplyr filter select
+#' @author PLP
+module_data_FAO_BilateralTrade <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(NULL)
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("FAO_BilateralTrade"))
+  } else if(command == driver.MAKE) {
+
+    Element <- Reporter.Countries <- Partner.Countries <-
+      Item.Code <- Item <- item.code <- NULL # silence package check.
+
+    # The bilateral trade dataset is massive (286m data points). The order of the steps below is intended to minimize
+    # processing time (i.e., the dataset is filtered before performing operations)
+
+    # Select columns, filter to the import and export quantity variables, and gather_years dropping NAs
+    # as the "matrix" is very sparse
+    fn <- "aglu/FAO/FAO_BilateralTrade"
+    fqfn <- find_csv_file(fn, FALSE, quiet = TRUE)
+    load_csv_files(fn, FALSE, quiet = TRUE)[[1]] %>%
```

Comment:
This is unusual; why is not happening via the usual `DECLARE_INPUTS` mechanism? Because we want to keep it in memory as short a time as possible? A (clearer) explanatory note might be useful.